public/js/modMap.js

```diff
@@ -63,53 +56,28 @@ function modMap() {
             width: 1
           })
         }
-        if (t["shape"] == "T3") {
-          if (t["orientation"] == "Ascend") {
-            var rect = draw.polygon('0,0 60,20 60,0').attr({
-              fill: defaultColor,
-              opacity: op,
-              x: i * 20,
-              y: n * 60
-            }).data('key', {
-              x,
-              y
-            }).stroke({
-              color: '#5ECCA2',
-              width: 1
-            })
+        if (mod['shape'] === 'T3') {
+          const {flipped} = mod;
+          const topLeft = `${i * 20},${n * 60}`;
+          const topRight = `${(i+1) * 20},${n * 60}`;
+          const bottomLeft = `${(i) * 20},${(n+1) * 60}`;
+          const bottomRight = `${(i+1) * 20},${(n+1) * 60}`;
 
-          }
-          if (t["orientation"] == "Decend") {
-            var rect = draw.polygon('0,0 60,20 60,0').attr({
-              fill: defaultColor,
-              opacity: op,
-              x: i * 20,
-              y: n * 60
-            }).data('key', {
-              x,
-              y
-            }).stroke({
-              color: '#5ECCA2',
-              width: 1
-            })
+          let coordinates;
+
+          if (mod['orientation'] === 'RH') {
+            coordinates = [topRight, bottomLeft];
+
+            coordinates.push(flipped ? bottomRight : topLeft);
+          } else if (mod['orientation'] === 'LH') {
+            coordinates = [topLeft, bottomRight];
 
+            coordinates.push(flipped ? topRight : bottomLeft);
           }
-          var rect = draw.rect(20, 60).attr({
-            fill: defaultColor,
-            opacity: op,
-            x: i * 20,
-            y: n * 60
-          }).data('key', {
-            x,
-            y
-          }).stroke({
-            color: '#5ECCA2',
-            width: 1
-          })
 
-        }
-        if (t["shape"] == "R3") {
-          var rect = draw.rect(20, 60).attr({
+          coordinates = coordinates.join(' ');
```

Comment:
I got an error here when a user has a `T3` with the orientation `flat` the page wouldn't render. Can you add something like this so it still renders but it shows the user is an idiot
```
          if (mod['orientation'] === 'flat') {
            coordinates = [topRight, bottomLeft];
            coordinates.push(flipped ? bottomRight : topLeft);
            let defaultColor = '#C73316';
```

Reply:
what does `flat` mean?

Reply:
Flat is for rectangles

Reply:
dd2dd0a